modules/nextflow/src/main/groovy/nextflow/Channel.groovy

```diff
@@ -657,4 +660,42 @@ class Channel  {
         fromPath0Future = future.exceptionally(Channel.&handlerException)
     }
 
-}
+    static DataflowWriteChannel fromLineage(String uri) {
+        final result = CH.create()
+        if( NF.isDsl2() ) {
+            session.addIgniter { fromLineage0(result, uri) }
+        }
+        else {
+            fromLineage0(result, uri )
+        }
+        return result
+    }
+
+    private static void fromLineage0(DataflowWriteChannel channel, String uri) {
+        final operation = Plugins.getExtension(LinChannelEx)
+        if( !operation )
+            throw new IllegalStateException("Unable to load lineage extensions.")
+        def future = CompletableFuture.runAsync( { operation.viewLineage(session, channel, new URI(uri)) } as Runnable)
+        future.exceptionally(this.&handlerException)
+    }
+
+    static DataflowWriteChannel queryLineage(String queryString) {
```

Comment:
Let's just use a Map here instead of query string. I would apply the same change to the `find` command as well. There is no need to add the extra complexity of URL encoding